src/SymmetricIntervalHull.jl

```diff
@@ -0,0 +1,238 @@
+
+### Output
+
+The origin.
+"""
+function center(sih::SymmetricIntervalHull{N})::Vector{N} where {N<:Real}
+    return zeros(N, dim(sih))
+end
+
+
+# --- LazySet interface functions ---
+
+
+"""
+    dim(sih::SymmetricIntervalHull)::Int
+
+Return the dimension of a symmetric interval hull of a convex set.
+
+### Input
+
+- `sih` -- symmetric interval hull of a convex set
+
+### Output
+
+The ambient dimension of the symmetric interval hull of a convex set.
+"""
+function dim(sih::SymmetricIntervalHull)::Int
+    return dim(sih.X)
+end
+
+"""
+    σ(d::AbstractVector{N}, sih::SymmetricIntervalHull
+     )::AbstractVector{<:Real} where {N<:Real}
+
+Return the support vector of a symmetric interval hull of a convex set in a
+given direction.
+
+### Input
+
+- `d`   -- direction
+- `sih` -- symmetric interval hull of a convex set
+
+### Output
+
+The support vector of the symmetric interval hull of a convex set in the given
+direction.
+If the direction has norm zero, the origin is returned.
+
+### Algorithm
+
+For each non-zero entry in `d` we need to either look up the bound (if it has
+been computed before) or compute it, in which case we store it for future
+queries.
+One such computation just asks for the support vector of the underlying set for
+both the positive and negative unit vector in the respective dimension.
+"""
+function σ(d::AbstractVector{N}, sih::SymmetricIntervalHull
+          )::AbstractVector{<:Real} where {N<:Real}
+    len = length(d)
+    @assert len == dim(sih) "cannot compute the support vector of a " *
+        "$(dim(sih))-dimensional set along a vector of length $(length(d))"
+
+    svec = similar(d)
+    zero_N = zero(N)
+    for i in eachindex(d)
+        if d[i] == zero_N
+            svec[i] = zero_N
+        else
+            if !sih.known[i]
+                compute_radius(sih, i, len)
+            end
+            svec[i] = sign(d[i]) * sih.radius[i]
+        end
+    end
+    return svec
+end
+
+
+# --- SymmetricIntervalHull functions ---
+
+
+"""
+    compute_radius(sih::SymmetricIntervalHull, i::Int, [dimension]::Int=dim(sih))
+
+Compute the radius of a symmetric interval hull of a convex in a given
+dimension.
+
+### Input
+
+- `sih`       -- symmetric interval hull of a convex set
+- `i`         -- dimension
+- `dimension` -- (optional, default: `dim(sih)`) set dimension
+
+### Output
+
+The radius of the symmetric interval hull of a convex set in the given
+direction.
+
+### Algorithm
+
+We ask for the support vector of the underlying set for both the positive and
+negative unit vector in the dimension `i`.
+"""
+function compute_radius(sih::SymmetricIntervalHull,
+                        i::Int,
+                        dimension::Int=dim(sih))
+    one_N = one(N)
+    right_bound = σ(sparsevec([i], [one_N], dimension), sih.X)
+    left_bound = σ(sparsevec([i], [-one_N], dimension), sih.X)
+    sih.radius[i] = max(right_bound[i], abs(left_bound[i]))
+    sih.known[i] = true
```

Comment:
should we invert these last two lines? such that it returns the computed radius which is more interesting.

Reply:
Actually, the docs said that it returns the radius...